compositor_pipeline/src/pipeline/pipeline_output.rs

```diff
@@ -107,6 +112,73 @@ impl Pipeline {
 
         Ok(port)
     }
+
+    pub(super) fn get_output_video_sender(
+        pipeline: &Arc<Mutex<Pipeline>>,
+        output_id: &OutputId,
+    ) -> Option<Sender<PipelineEvent<Frame>>> {
+        let mut guard = pipeline.lock().unwrap();
+        let Some(output) = guard.outputs.get_mut(output_id) else {
+            warn!(
+                ?output_id,
+                "Failed to send output frame. Output does not exists.",
+            );
```

Comment:
If I understand correctly, those logs can happen even if there is no error. In the case of output unregistering, outputs may be unregistered in the middle of the rendering/mixing, because mixer/renderer are working on a different threads than the pipeline, causing those logs to appear. Since this is a warning, not an error, I don't have a strong opinion on that, but it seems weird to display warn logs in perfectly normal situations. I noticed this kind of log once while working on demos.

Reply:
Yes, I prefer to keep this warning. If we want to avoid false positives we could call this method before render on all outputs. But this would requires some refactor, because if we take that approach we can't always unregister_output when this function returns None